Sources/Core/AWSSDKIdentity/Sources/InternalAWSSTS/IdentityProvidingSTSClient.swift

```diff
@@ -8,28 +8,40 @@
 // Code generated by smithy-swift-codegen. DO NOT EDIT!
 
 import Foundation
+import class AWSSDKIdentity.CredentialFeatureIDInterceptorProvider
 import enum AWSSDKIdentity.AWSCredentialIdentityResolverError
+import enum AWSSDKIdentity.AWSIdentityPropertyKeys
+import enum AWSSDKIdentity.CredentialFeatureID
 import enum AWSSDKIdentity.IdentityProvidingSTSClientError
 import protocol AWSSDKIdentity.IdentityProvidingSTSClient
 import struct AWSSDKIdentity.AWSCredentialIdentity
+import struct Smithy.Attributes
 import struct SmithyIdentity.StaticAWSCredentialIdentityResolver
 
 package struct IdentityProvidingSTSClient: AWSSDKIdentity.IdentityProvidingSTSClient, Swift.Sendable {
     package init() {}
 
-    package func assumeRoleWithCreds(creds: AWSSDKIdentity.AWSCredentialIdentity, roleARN: String, roleSessionName: String, durationSeconds: Foundation.TimeInterval) async throws -> AWSSDKIdentity.AWSCredentialIdentity {
+    package func assumeRoleWithCreds(creds: AWSSDKIdentity.AWSCredentialIdentity, roleARN: String, roleSessionName: String, durationSeconds: Foundation.TimeInterval, credentialFeatureIDs: [String]) async throws -> AWSSDKIdentity.AWSCredentialIdentity {
         let stsConfig = try await STSClient.STSClientConfiguration()
         stsConfig.awsCredentialIdentityResolver = SmithyIdentity.StaticAWSCredentialIdentityResolver(creds)
         let sts = STSClient(config: stsConfig)
         let out = try await sts.assumeRole(input: AssumeRoleInput(durationSeconds: Int(durationSeconds), roleArn: roleARN, roleSessionName: roleSessionName))
         guard let creds = out.credentials, let accessKey = creds.accessKeyId, let secretKey = creds.secretAccessKey else {
             throw AWSSDKIdentity.AWSCredentialIdentityResolverError.failedToResolveAWSCredentials("STSAssumeRoleAWSCredentialIdentityResolver:Failed to retrieve credentials from STS with assume role.")
         }
-        return AWSCredentialIdentity(accessKey: accessKey, secret: secretKey, expiration: creds.expiration, sessionToken: creds.sessionToken)
+        var properties = Smithy.Attributes()
+        if credentialFeatureIDs.last == AWSSDKIdentity.CredentialFeatureID.CREDENTIALS_STS_ASSUME_ROLE.rawValue {
+            properties.set(key: AWSSDKIdentity.AWSIdentityPropertyKeys.credentialFeatureIDs, value: credentialFeatureIDs)
+        } else {
+            properties.set(key: AWSSDKIdentity.AWSIdentityPropertyKeys.credentialFeatureIDs, value: credentialFeatureIDs + [AWSSDKIdentity.CredentialFeatureID.CREDENTIALS_STS_ASSUME_ROLE.rawValue])
+        }
+        return AWSCredentialIdentity(accessKey: accessKey, secret: secretKey, expiration: creds.expiration, sessionToken: creds.sessionToken, properties: properties)
```

Comment:
Logic here determines whether to append `"i"` credential feature ID into newly resolved AWS credential object's list of credential feature IDs or not. `"i"` marks successful completion of assumeRole call to resolve credentials. The reason why it skips adding anything if it already ends in `"i"` is to discard duplicates. The reason why skipping works is because if list of feature IDs resolved so far already ends with `i`, that means we're at the point of simply recursing back up the `source_profile` chain (i.e., only remaining work is a bunch of assumeRole calls, as many as however many `source_profile` nesting there was), and only feature IDs that will be appended going forward is just a bunch of `i`'s.